x/rewards/keeper/iprpc.go

```diff
@@ -148,6 +147,26 @@ func (k Keeper) distributeIprpcRewards(ctx sdk.Context, iprpcReward types.IprpcR
 			continue
 		}
 
+		// remove providers that are registered in the spec fund but unstaked (can't reward unstaked providers)
+		epoch, _, err := k.epochstorage.GetEpochStartForBlock(ctx, uint64(ctx.BlockHeight()))
```

Comment:
you dont need this, use the current stake storage, not for a specific epoch

Reply:
also i would advise to do this check when building the `specCuMap` map

Reply:
done. see https://github.com/lavanet/lava/pull/1678/commits/0d7f04963d0c02fbc37cc820106d8b2b1afa509b